vivarium/simulator/states.py

```diff
@@ -0,0 +1,316 @@
+    behavior: util.Array
+    wheel_diameter: util.Array
+    speed_mul: util.Array
+    theta_mul: util.Array
+    proxs_dist_max: util.Array
+    proxs_cos_min: util.Array
+    color: util.Array
+
+
+@dataclass
+class ObjectState:
+    nve_idx: util.Array  # idx in NVEState
+    color: util.Array
+
+
+# TODO : I think it would make more sense to have max_agents, max_objects here instead of n_***
+@dataclass
+class SimulatorState:
+    idx: util.Array
+    box_size: util.Array
+    n_agents: util.Array
+    n_objects: util.Array
+    num_steps_lax: util.Array
+    dt: util.Array
+    freq: util.Array
+    neighbor_radius: util.Array
+    to_jit: util.Array
+    use_fori_loop: util.Array
+
+    @staticmethod
+    def get_type(attr):
+        if attr in ['idx', 'n_agents', 'n_objects', 'num_steps_lax']:
+            return int
+        elif attr in ['box_size', 'dt', 'freq', 'neighbor_radius']:
+            return float
+        elif attr in ['to_jit', 'use_fori_loop']:
+            return bool
+        else:
+            raise ValueError()
+     
+
+@dataclass
+class State:
+    simulator_state: SimulatorState
+    nve_state: NVEState
+    agent_state: AgentState
+    object_state: ObjectState
+
+    def field(self, stype_or_nested_fields):
+        if isinstance(stype_or_nested_fields, StateType):
+            name = stype_or_nested_fields.name.lower()
+            nested_fields = (f'{name}_state', )
+        else:
+            nested_fields = stype_or_nested_fields
+
+        res = self
+        for f in nested_fields:
+            res = getattr(res, f)
+
+        return res
+
+    # TODO : Should we keep this function because it is duplicated below ? 
+    # def nve_idx(self, etype):
+    #     cond = self.e_cond(etype)
+    #     return compress(range(len(cond)), cond)  # https://stackoverflow.com/questions/21448225/getting-indices-of-true-values-in-a-boolean-list
+
+    def nve_idx(self, etype, entity_idx):
+        return self.field(etype).nve_idx[entity_idx]
+
+    def e_idx(self, etype):
+        return self.nve_state.entity_idx[self.nve_state.entity_type == etype.value]
+
+    def e_cond(self, etype):
+        return self.nve_state.entity_type == etype.value
+
+    def row_idx(self, field, nve_idx):
+        return nve_idx if field == 'nve_state' else self.nve_state.entity_idx[jnp.array(nve_idx)]
+
+    def __getattr__(self, name):
+        def wrapper(e_type):
+            value = getattr(self.nve_state, name)
+            if isinstance(value, rigid_body.RigidBody):
+                return rigid_body.RigidBody(center=value.center[self.e_cond(e_type)],
+                                            orientation=value.orientation[self.e_cond(e_type)])
+            else:
+                return value[self.e_cond(e_type)]
+        return wrapper
+
+
+# Helper function to transform a color string into rgb with matplotlib colors
+def _string_to_rgb(color_str):
+    return jnp.array(list(mcolors.to_rgb(color_str)))
+
+
+def init_simulator_state(
```

Comment:
We can do this in another Issue/PR, but it would be better if default values are defined at a single place in the code. For now we have them in several places, e.g. here, in the run scripts, in the Config classes (and maybe elsewhere), this will be very hard to maintain. We could e.g. have them in a dedicated file and all the classes/functions that need them would import them from there. Might be easier to do it in the context of #30 though (if you agree please update #30 or related PR so that we don't forget, otherwise let's discuss it this afternoon)